pallets/liquidity-pools-gateway/src/message_processing.rs

```diff
@@ -293,24 +293,24 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// Gets the message proof for a message.
-	pub(crate) fn get_message_proof(message: T::Message) -> Proof {
-		match message.get_proof() {
+	pub(crate) fn get_message_hash(message: T::Message) -> MessageHash {
+		match message.get_message_hash() {
 			None => message
 				.to_proof_message()
-				.get_proof()
-				.expect("message proof ensured by 'to_message_proof'"),
+				.get_message_hash()
+				.expect("message hash ensured by 'to_proof_message'"),
 			Some(proof) => proof,
 		}
```

Comment:
Really like the new name, but not sure if this reading process makes sense now here. Putting me in the mind of an auditor:

- I get the hash of a message
- If the message has no hash: (<- 🤔 could a message not have a hash?)
  - We create a proof of a message (<- 🤔  Why do I need to create a proof message to get the hash?)
  - And then we get the hash

Reply:
Yea, it might be a bit confusing, but I dunno how to solve this without adding another method to the trait like `has_message_hash`, then still do a call and do an `unwrap/expect` on it.

Reply:
I'll update this in a bit and ping you.